openmc/plots.py

```diff
@@ -261,6 +261,97 @@ def __repr__(self):
         string += '{0: <16}{1}{2}\n'.format('\tCol Spec', '=\t', self._col_spec)
         return string
 
+    def colorize(self, geometry, seed=1):
+        """Generate a color scheme for each domain in the plot.
+
+        This routine may be used to generate random, reproducible color schemes.
+        The colors generated are based upon cell/material IDs in the geometry.
+
+        Params
+        ------
+        geometry : openmc.Geometry
+            The geometry for which the plot is defined
+        seed : Integral
+            The random number seed used to generate the color scheme
+
+        """
+        
+        cv.check_type('geometry', geometry, openmc.Geometry)
+        cv.check_type('seed', seed, Integral)
+        cv.check_greater_than('seed', seed, 1, equality=True)
+
+        # Get collections of the domains which will be plotted
+        if self.color is 'mat':
+            domains = geometry.get_all_materials()
+        else:
+            domains = geometry.get_all_cells()
+
+        # Set the seed for the random number generator
+        np.random.seed(seed)
+
+        # Generate random colors for each feature
+        self.col_spec = {}
+        for domain in domains:
+            r = np.random.randint(0, 255)
+            g = np.random.randint(0, 255)
+            b = np.random.randint(0, 255)
+            self.col_spec[domain] = (r, g, b)
+
+
+    def highlight_domains(self, geometry, domains, seed=1,
+                          alpha=0.5, background='grey'):
+        """Use alpha compositing to highlight one or more domains in the plot.
+
+        This routine generates a color scheme and applies alpha compositing
+        to make all domains except the highlighted ones partially transparent.
```

Comment:
You know there's not much wrong with your code once I start picking at your English!